provider/pkg/ssa/actions.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ssa
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/yaml"
+)
+
+// Relinquish is used to "undo" a patch operation by relinquishing management of all fields in a resource for the
+// specified field manager. Any fields that no longer have a manager will return to a default value, so this isn't
+// guaranteed to return the resource back to the same state that it was in before the patch.
+//
+// This operation will fail unless another manager is responsible for a valid resource spec. In other words, if the
+// specified field manager is the sole manager of a resource, then this will fail with a validation error for most
+// resources.
+func Relinquish(
+	ctx context.Context,
+	client dynamic.ResourceInterface,
+	inputs *unstructured.Unstructured,
```

Comment:
Passing the entire inputs was pretty confusing for me and I initially parsed this to suggest we were using the inputs as the body for the patch (which wouldn't do the relinquishing at all). Could you replace this with just the name so it makes it explicit how its being used?

Reply:
We chatted about this separately, but I realized that I'm also using the inputs to populate the minimal spec (GVK + namespace/name). It's simpler to leave as is, but I can clarify the usage with a comment.